File: Csharp/Visitor/Visitor/Visitor2.cs
using System;
namespace Show
{
    class Visitor2: IVisitor
    {
        public void VisitComponentA(ComponentA element)
        {
            Console.WriteLine(element.ExclusiveMethodOfComponentA() + " + Visitor2");
        }

        public void VisitComponentB(ComponentB element)
        {
            Console.WriteLine(element.SpecialMethodOfComponentB() + " + Visitor2");
        }
    }
}

File: PHP/Behavioral/Visitor/V1/Visitor2.php
<?php

class Visitor2 implements Visitor
{
    public function visitComponentA(ComponentA $element): void
    {
        echo $element->exclusiveMethodOfComponentA() . " + Visitor2<br/>";
    }

    public function visitComponentB(ComponentB $element): void
    {
        echo $element->specialMethodOfComponentB() . " + Visitor2<br/>";
    }
}